File: app/src/main/java/com/latitude/seventimer/util/L.java
package com.latitude.seventimer.util;

import android.util.Log;

import androidx.annotation.NonNull;

public class L {
    public static boolean logSwitch = true;
    public static int logLevel = 1;
    private static String version = "";
    private static int LOG_MAXLENGTH = 2000;

    public static void e(String tag, String msg) {
        if (logLevel <= Log.ERROR || logSwitch) {
            Log.e(version + tag, msg);
        }

    }

    public static void d(String tag, String msg) {
        if (logLevel <= Log.DEBUG || logSwitch) {
            Log.d(version + tag, msg);
        }
    }

    public static void d(String tag, String format, Object... args) {
        d(tag, formatMessage(format, args));
    }

    public static void w(String tag, String msg) {
        if (logLevel <= Log.WARN || logSwitch) {
            Log.w(version + tag, msg);
        }
    }

    public static void i(String tag, String msg) {
        if (logLevel <= Log.INFO || logSwitch) {
            Log.i(version + tag, msg);
        }

    }

    public static void v(String tag, String msg) {
        if (logLevel <= Log.VERBOSE || logSwitch) {
            Log.v(version + tag, msg);
        }
    }

    public static void LongString(String tag, String msg) {
        if (logLevel <= Log.VERBOSE || logSwitch) {

            int strLength = msg.length();
            int start = 0;
            int end = LOG_MAXLENGTH;
            for (int i = 0; i < 100; i++) {
                if (strLength > end) {
                    L.e(tag, i + ": " + msg.substring(start, end));
                    start = end;
                    end = end + LOG_MAXLENGTH;
                } else {
                    L.e(tag, i + ": " + msg.substring(start, strLength));
                    break;
                }
            }
        }
    }

    private static String formatMessage(@NonNull String message, @NonNull Object[] args) {
        return String.format(message, args);
    }

}

File: app/src/main/java/com/latitude/seventimer/util/L.kt
package com.latitude.seventimer.util

import android.util.Log

object L {

    const val logSwitch = true
    const val logLevel = 1
    private val version = ""
    private val LOG_MAXLENGTH = 2000

    @JvmStatic
    fun e(tag: String, msg: String) {
        if (logLevel <= Log.ERROR || logSwitch) {
            Log.e(version + tag, msg)
        }
    }

    fun e(tag: String, format: String, vararg args: Any) {
        e(tag, formatMessage(format, *args))
    }

    @JvmStatic
    fun d(tag: String, msg: String) {
        if (logLevel <= Log.DEBUG || logSwitch) {
            Log.d(version + tag, msg)
        }
    }

    @JvmStatic
    fun d(tag: String, format: String, vararg args: Any) {
        d(tag, formatMessage(format, *args))
    }

    @JvmStatic
    fun w(tag: String, msg: String) {
        if (logLevel <= Log.WARN || logSwitch) {
            Log.w(version + tag, msg)
        }
    }

    @JvmStatic
    fun w(tag: String, format: String, vararg args: Any) {
        w(tag, formatMessage(format, *args))
    }

    @JvmStatic
    fun i(tag: String, msg: String) {
        if (logLevel <= Log.INFO || logSwitch) {
            Log.i(version + tag, msg)
        }
    }

    @JvmStatic
    fun i(tag: String, format: String, vararg args: Any) {
        i(tag, formatMessage(format, *args))
    }

    @JvmStatic
    fun v(tag: String, msg: String) {
        if (logLevel <= Log.VERBOSE || logSwitch) {
            Log.v(version + tag, msg)
        }
    }

    @JvmStatic
    fun v(tag: String, format: String, vararg args: Any) {
        v(tag, formatMessage(format, *args))
    }

    @JvmStatic
    fun LongString(tag: String, msg: String) {
        if (logLevel <= Log.VERBOSE || logSwitch) {

            val strLength = msg.length
            var start = 0
            var end = LOG_MAXLENGTH
            for (i in 0..99) {
                if (strLength > end) {
                    L.e(tag, i.toString() + ": " + msg.substring(start, end))
                    start = end
                    end = end + LOG_MAXLENGTH
                } else {
                    L.e(tag, i.toString() + ": " + msg.substring(start, strLength))
                    break
                }
            }
        }
    }

    fun formatMessage(message: String, vararg args: Any): String {
        return String.format(message, *args)
    }

}
